streams/src/main/java/org/apache/kafka/streams/internals/metrics/OpenIterators.java

```diff
@@ -50,24 +48,22 @@ public OpenIterators(final TaskId taskId,
 
     public void add(final MeteredIterator iterator) {
         openIterators.add(iterator);
-        numOpenIterators.increment();
 
-        if (numOpenIterators.intValue() == 1) {
+        if (openIterators.size() == 1) {
```

Comment:
I was worried for adding the same `iterator`, which could create unexpected metrics, but it seems `StateStoreMetrics.addOldestOpenIteratorGauge` already handle this case.

Reply:
Not sue if I can follow. Can you elaborate?

Reply:
Sorry for the unclear comment. Please see the following example

```java
        var openIterators = new OpenIterators();
        var iterator = new MeteredIterator(1000L);
        openIterators.add(iterator);
        openIterators.add(iterator);
```

The second `openIterators.add(iterator)` did not invoke `StateStoreMetrics.addOldestOpenIteratorGauge` before, but it does now. However, this should be fine, since `StateStoreMetrics.addOldestOpenIteratorGauge` does not create new metrics in this scenario.

Reply:
Well, would be a bug anyway, to call `openIterators.add(iterator);` twice on the same `iterator` object. Each iterator should be registered exactly one time.